lib/charms/haproxy/v0/haproxy_route.py

```diff
@@ -0,0 +1,1191 @@
+
+        Args:
+            service: The name of the service to route traffic to.
+            ports: List of ports the service is listening on.
+            paths: List of URL paths to route to this service.
+            subdomains: List of subdomains to route to this service.
+            check_interval: Interval between health checks in seconds.
+            check_rise: Number of successful health checks before server is considered up.
+            check_fall: Number of failed health checks before server is considered down.
+            check_paths: List of paths to use for health checks.
+            path_rewrite_expressions: List of regex expressions for path rewrites.
+            query_rewrite_expressions: List of regex expressions for query rewrites.
+            header_rewrite_expressions: List of tuples containing header name
+                and rewrite expression.
+            load_balancing_algorithm: Algorithm to use for load balancing.
+            load_balancing_cookie: Cookie name to use when algorithm is set to cookie.
+            rate_limit_connections_per_minute: Maximum connections allowed per minute.
+            rate_limit_policy: Policy to apply when rate limit is reached.
+            upload_limit: Maximum upload bandwidth in bytes per second.
+            download_limit: Maximum download bandwidth in bytes per second.
+            retry_count: Number of times to retry failed requests.
+            retry_interval: Interval between retries in seconds.
+            retry_redispatch: Whether to redispatch failed requests to another server.
+            deny_paths: List of paths that should not be routed to the backend.
+            server_timeout: Timeout for requests from haproxy to backend servers in seconds.
+            client_timeout: Timeout for client requests to haproxy in seconds.
+            queue_timeout: Timeout for requests waiting in queue in seconds.
+            server_maxconn: Maximum connections per server.
+        """
+        self._application_data = self._generate_application_data(
+            service,
+            ports,
+            paths,
+            subdomains,
+            check_interval,
+            check_rise,
+            check_fall,
+            check_paths,
+            path_rewrite_expressions,
+            query_rewrite_expressions,
+            header_rewrite_expressions,
+            load_balancing_algorithm,
+            load_balancing_cookie,
+            rate_limit_connections_per_minute,
+            rate_limit_policy,
+            upload_limit,
+            download_limit,
+            retry_count,
+            retry_interval,
+            retry_redispatch,
+            deny_paths,
+            server_timeout,
+            client_timeout,
+            queue_timeout,
+            server_maxconn,
+        )
+        self.update_relation_data()
+
+    # pylint: disable=too-many-arguments,too-many-positional-arguments,too-many-locals
+    def _generate_application_data(
```

Comment:
Due to the way python works, we should probably not set any default value here, and add checks in the code `if value is None: value = "default"`

Reply:
removed the complex lambda logic and made it simple and straightforward